### PR TITLE
apiextension: fix typo and test case in conversion integration test

### PR DESCRIPTION
**Pull request by sttts**
Replaces #96967.

/kind bug
/kind failing-test

```release-note
NONE
```

### CODE REVIEW

**Comment by sttts** [comment]
/assign @liggitt 

cc @bl-ue

**Review by bl-ue — APPROVED** [review]
Cool — thank you for fixing this, and I'm sorry that I couldn't ;)

File: staging/src/k8s.io/apiextensions-apiserver/test/integration/conversion/conversion_test.go [hunk 1]
```diff
@@ -687,7 +693,7 @@ func expectConversionFailureMessage(id, message string) func(t *testing.T, ctc *
 				case "create":
 					_, err = clients["v1beta2"].Create(context.TODO(), newConversionMultiVersionFixture(ns, id, "v1beta2"), metav1.CreateOptions{})
 				case "update":
-					_, err = clients["v1beta2"].Update(context.TODO(), obj, metav1.UpdateOptions{})
+					_, err = clients["v1beta2"].Update(context.TODO(), objv1beta2, metav1.UpdateOptions{})
 				case "patch":
 					_, err = clients["v1beta2"].Patch(context.TODO(), obj.GetName(), types.MergePatchType, []byte(`{"metadata":{"annotations":{"patch":"true"}}}`), metav1.PatchOptions{})
 				case "delete":
```
Comment (liggitt) [inline staging/src/k8s.io/apiextensions-apiserver/test/integration/conversion/conversion_test.go:699]: looks good... also add a `default:` case that errors to catch any other ineffective loop values

Comment (liggitt) [reply staging/src/k8s.io/apiextensions-apiserver/test/integration/conversion/conversion_test.go:699]: same issue exists below with with subresource looping over `udpate`... add the default cases to the switch statements there as well

Comment (sttts) [reply staging/src/k8s.io/apiextensions-apiserver/test/integration/conversion/conversion_test.go:699]: done

**Pull request merged**